onnxscript/rewriter/pattern.py

```diff
@@ -1812,10 +1825,20 @@ def print(self):
                     print(f"Graph matching failed: {reason}")
             else:
                 print("Graph matching failed.")
-            failure_node = self.match_result._failure_node
-            if failure_node:
-                print("Failure at or around node:")
-                failure_node.display()
+            failure_nodes_and_values = self.match_result._failure_nodes_and_values
+            print("Failure at or around nodes/values:")
+
+            if failure_nodes_and_values:
+                if isinstance(failure_nodes_and_values, list):
```

Comment:
Don't we expect a list in all cases? However, I think we can special-case when the list has a single element (just to produce a slightly better wording of error message).